Helper/Order.php

```diff
@@ -1873,4 +1883,56 @@ public function getOrderStoreIdByDisplayId($displayId)
 
         return ($order && $order->getStoreId()) ? $order->getStoreId() : null;
     }
+
+    /**
+     * Add comment to status history if order doesn't have similar comment
+     *
+     * @param string $displayId Display id. Method find order by it
+     * @param string $comment comment we need to add
+     * @param string $searchPhrase We don't add comment if order already has any comment with this phrase
+     */
+    public function addCommentToStatusHistoryIfNotExists($displayId, $comment, $searchPhrase)
+    {
+        list($incrementId, $quoteId) = $this->getDataFromDisplayID($displayId);
+        $order = $this->getExistingOrder($incrementId);
+        if (!$order) {
+            return;
+        }
+
+        foreach ($order->getAllStatusHistory() as $orderComment) {
+            if (strpos($orderComment->getComment(), $searchPhrase) !== false) {
+                return;
+            }
+        }
+        $order->addCommentToStatusHistory($comment);
+        $order->save();
+    }
+
+    /**
+     * Subscribe for newsletters
+     * - Find order by displayId
+     * - If order for logged in user subscribe by userId
+     * - If order for guest user subscribe by email
+     *
+     * @param $displayId
+     */
+    public function subscribeForNewsletter($displayId) {
```

Comment:
@danac-gs What do you think about this method? We can move it into new checkboxes helper, or leave it in this helper, in case we need to subscribe in case not related to checkboxes (very hypothetically).
What do you prefer?

Reply:
@vitaliyreznikov I'd leave it here or more likely create a new helper class and put it there, \Bolt\Boltpay\Helper\Customer.
And change its parameter(s) from ($displayId) to either ($order) or even better ($customerId = null, $email = null) so we can (hypothetically) use it independently on the context it's been used.

Reply:
@vitaliyreznikov after taking another look I think it's OK to leave this method here but pass it the order (get it from display_id before entering the method). We probably won't have another context of using this than the order.